### PR TITLE
Extending XPQA to add cross lingual evaluation

### PR DESCRIPTION
**Pull request by wissam-sib**
The current XPQA was multilingual but it can be extended to be cross lingual.

- [x] I have tested that the dataset runs with the `mteb` package.
- [x] I have run the following models on the task (adding the results to the pr). These can be run using the `mteb run -m {model_name} -t {task_name}` command.
  - [x] `sentence-transformers/paraphrase-multilingual-MiniLM-L12-v2`
  - [x] `intfloat/multilingual-e5-small`
- [x] I have checked that the performance is neither trivial (both models gain close to perfect scores) nor random (both models gain close to random scores).
- [x] I have filled out the metadata object in the dataset file (find documentation on it [here](https://github.com/embeddings-benchmark/mteb/blob/main/docs/adding_a_dataset.md#2-creating-the-metadata-object)).
- [x] Run tests locally to make sure nothing is broken using `make test`. 
- [x] Run the formatter to format the code using `make lint`. 
- [x] I have added points for my submission to the [points folder](https://github.com/embeddings-benchmark/mteb/blob/main/docs/mmteb/points.md) using the PR number as the filename (e.g. `438.jsonl`).

### CODE REVIEW

File: mteb/tasks/Retrieval/multilingual/XPQARetrieval.py [hunk 1]
```diff
@@ -74,33 +69,107 @@ class XPQARetrieval(MultilingualTask, AbsTaskRetrieval):
...
+        text_creation="found",
+        bibtex_citation="""@inproceedings{shen2023xpqa,
+        title={xPQA: Cross-Lingual Product Question Answering in 12 Languages},
+        author={Shen, Xiaoyu and Asai, Akari and Byrne, Bill and De Gispert, Adria},
+        booktitle={Proceedings of the 61st Annual Meeting of the Association for Computational Linguistics (Volume 5: Industry Track)},
+        pages={103--115},
+        year={2023}
+        }""",
+        n_samples={"test": 19801},
+        avg_character_length={"test": 104.68},  # answer
```
Comment (KranthiGV) [inline mteb/tasks/Retrieval/multilingual/XPQARetrieval.py:92]: Now that this metadata is filled, we can remove this from `_HISTORIC_DATASETS` of `tests/test_TaskMetadata.py`

Comment (wissam-sib) [reply mteb/tasks/Retrieval/multilingual/XPQARetrieval.py:92]: will give it a try

File: mteb/tasks/Retrieval/multilingual/XPQARetrieval.py [hunk 2]
```diff
@@ -74,33 +69,107 @@ class XPQARetrieval(MultilingualTask, AbsTaskRetrieval):
...
+        dataset = dataset.filter(lambda x: x["answer"] is not None)
+        # making sure that the question is not in the context
+        dataset = dataset.map(
+            lambda example: {
+                "context": example["context"][
+                    example["context"].find("<strong>") + len("<strong>") : example[
+                        "context"
+                    ].rfind("</strong>")
+                ]
+            }
```
[the remark this answers is not in the document]
Comment (wissam-sib) [reply mteb/tasks/Retrieval/multilingual/XPQARetrieval.py:115]: In fact, in this dataset, the contexts are Amazon's QA (or something similar) so you often have a question and an answer in between the strong tags. But the question can be the same as the field question which makes it too easy.

File: mteb/tasks/Retrieval/multilingual/XPQARetrieval.py [hunk 3]
```diff
@@ -74,33 +69,107 @@ class XPQARetrieval(MultilingualTask, AbsTaskRetrieval):
...
 
+        self.queries, self.corpus, self.relevant_docs = {}, {}, {}
+        for lang_pair, _ in self.metadata.eval_langs.items():
+            lang_corpus, lang_question = lang_pair.split("-")
+            lang_not_english = lang_corpus if lang_corpus != "eng" else lang_question
+            dataset_language = dataset.filter(
+                lambda x: x["lang"] == _LANG_CONVERSION.get(lang_not_english)
+            )
+            question_key = "question_en" if lang_question == "eng" else "question"
+            corpus_key = "context" if lang_corpus == "eng" else "answer"
```
[the remark this answers is not in the document]
Comment (wissam-sib) [reply mteb/tasks/Retrieval/multilingual/XPQARetrieval.py:126]: In fact the context is almost always just the answer so this doesn't make it really inconsistent. In fact, the performance is actually lower with eng as corpus. Will look at the performance without this transform:
```
dataset.map(
            lambda example: {
                "context": example["context"][
                    example["context"].find("<strong>") + len("<strong>") : example[
                        "context"
                    ].rfind("</strong>")
                ]
            }
```

And see if more symmetrical

File: mteb/tasks/Retrieval/multilingual/XPQARetrieval.py (hunk 2 above)
[the remark this answers is not in the document]
Comment (wissam-sib) [reply mteb/tasks/Retrieval/multilingual/XPQARetrieval.py]: ended up using another column called "candidate" which is more relevant for symmetry

File: mteb/tasks/Retrieval/multilingual/XPQARetrieval.py (hunk 3 above)
[the remark this answers is not in the document]
Comment (wissam-sib) [reply mteb/tasks/Retrieval/multilingual/XPQARetrieval.py]: ended up using another column called "candidate" which is more relevant for symmetry

[the remark this answers is not in the document]
Comment (imenelydiaker) [reply mteb/tasks/Retrieval/multilingual/XPQARetrieval.py]: so candidate is equivalent to answer for all languages?

File: mteb/tasks/Retrieval/multilingual/XPQARetrieval.py [hunk 4]
```diff
@@ -74,33 +71,107 @@ class XPQARetrieval(MultilingualTask, AbsTaskRetrieval):
...
+            lang_corpus, lang_question = (
+                lang_pair.split("-")[0],
+                lang_pair.split("-")[1],
+            )
+            lang_not_english = lang_corpus if lang_corpus != "eng" else lang_question
+            dataset_language = dataset.filter(
+                lambda x: x["lang"] == _LANG_CONVERSION.get(lang_not_english)
+            )
+            question_key = "question_en" if lang_question == "eng" else "question"
+            corpus_key = "candidate" if lang_corpus == "eng" else "answer"
```
Comment (imenelydiaker) [inline mteb/tasks/Retrieval/multilingual/XPQARetrieval.py:117]: so candidate and answer are quivalent for all languages?

Comment (wissam-sib) [reply mteb/tasks/Retrieval/multilingual/XPQARetrieval.py:117]: yes more or less, with candidate being in english and answer in the other language

**Comment by KennethEnevoldsen** [comment]
@imenelydiaker I am going through PRs. Will you have the time to finish up this PR?

**Comment by imenelydiaker** [comment]
> @imenelydiaker I am going through PRs. Will you have the time to finish up this PR?

Yes I'm on it 🙂

**Comment by KennethEnevoldsen** [comment]
Wonderful thanks!

**Review by imenelydiaker — APPROVED** [review]
Please add points @wissam-sib, thank you for this additon! 🙂

**Pull request merged**